### ISSUE TITLE
Rustdoc warns about comments in code blocks

### ISSUE THREAD

**Issue by macmv** [rust-lang/rust#96233]
Given the following code: 

```rust
/// ```rust
/// // Hello!
/// ```
fn main() {}
```

The current output of rustdoc is:

```
$ rustdoc src/main.rs --document-private-items
warning: Rust code block is empty
 --> src/main.rs:1:5
  |
1 |   /// ```rust
  |  _____^
2 | | /// // Hello!
3 | | /// ```
  | |_______^
  |
  = note: `#[warn(rustdoc::invalid_rust_codeblocks)]` on by default

warning: `rustdoc-sample` (bin "rustdoc-sample" doc) generated 1 warning
```

Ideally, the output should not give any warnings.

While a standalone comment in a code block is not a common situation, the resulting webpage will highlight it as a comment. I have a code generator, which creates rust code for build scripts. There is a function to generate a comment, and part of the documentation is the resulting comment. Using ```text would solve the warning, but it would not be highlighted as a comment in the resulting webpage. As far as I can tell, this doesn't seem like an empty code block, and I don't think this lint should trigger over it.

Using stable and switching editions doesn't change the output.

Here is a link to a sample repository: [https://github.com/macmv/rustdoc-empty-code-block-sample](https://github.com/macmv/rustdoc-empty-code-block-sample)

**Comment by GuillaumeGomez** [comment]
The issue can be simply checked with:

```rust
/// ```
/// // hello!
/// ```
pub fn foo() {}
```

Not particular option. And the code block is empty. So I'm not sure if there is anything to be done in here...

**Comment by macmv** [comment]
That still produces a warning. I'll just add `#[allow(rustdoc::invalid_rust_codeblocks)]`, and leave it. Using ```text will fix the warning, but then it won't look like a comment in the webpage.

**Comment by GuillaumeGomez** [comment]
Let's not close it. We're discussing it in [zulip](https://rust-lang.zulipchat.com/#narrow/stream/266220-rustdoc/topic/don't.20warn.20if.20a.20code.20block.20only.20contains.20comments.3F) so don't hesitate to come provide your point of view as well. We dont really know what do in here. :laughing:
